modules/oe_webtools_media/src/Plugin/Validation/Constraint/WebtoolsMediaConstraintValidator.php

```diff
@@ -58,4 +71,45 @@ public function validate($value, Constraint $constraint) {
     }
   }
 
+  /**
+   * Attempts to parse and validate a Wcloud snippet.
+   *
+   * @param array $snippet
+   *   The snippet to ve parsed and validated.
+   * @param \Symfony\Component\Validator\Constraint $constraint
+   *   The constraint object.
+   *
+   * @return array
+   *   The parsed snippet if validation passed or an empty array otherwise.
+   */
+  protected function parseWcloud(array $snippet, Constraint $constraint): array {
+    // Assert if the url property is set and is valid.
+    if (empty($snippet['url']) || !UrlHelper::isValid($snippet['url'], TRUE)) {
+      $this->context->addViolation($constraint->incorrectUrlMessage);
+      return [];
+    }
+    // Assert that the url contains the europa.eu domain.
+    if (strpos(parse_url($snippet['url'], PHP_URL_HOST), 'europa.eu') === FALSE) {
```

Comment:
Shouldn't we check that the domain ENDS in europa.eu? Otherwise I believe `miao.europa.eunot.com` might work.